### PR TITLE
Introduce deploy delegation to Spark batch

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spark/tools/job/AzureBlobStorageDeploy.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    public static CloudBlobContainer getBlobContainer(final CloudStorageAccount storageAccount,
+                                                      final String containerName)
+            throws URISyntaxException, StorageException {
+        final CloudBlobClient blobClient = storageAccount.createCloudBlobClient();
+        return blobClient.getContainerReference(containerName);
+    }
+
+    public static String getHadoopBlobFsPropertyKey(final String storageAccountName, final String endpointSuffix) {
```
Comment (t-rufang) [inline src/main/java/com/microsoft/azure/spark/tools/job/AzureBlobStorageDeploy.java]: Maybe we can put this method under class `WasbUri`

File: src/main/java/com/microsoft/azure/spark/tools/http/HttpObservable.java [hunk 2]
```diff
@@ -335,20 +338,20 @@ public HttpObservable setTrustStrategy(final TrustStrategy externalTrustStrategy
...
      * Helper to convert the http response to a specified type.
      *
+     * @param <T> the target type
      * @param resp HTTP response, consumed as String content
      * @param clazz the target type to convert
-     * @param <T> the target type
      * @return the specified type class instance
      */
-    public <T> T convertJsonResponseToObject(final HttpResponse resp, final Class<T> clazz) {
+    public <T> Pair<T, HttpResponse> convertJsonResponseToObject(final HttpResponse resp, final Class<T> clazz) {
```
Comment (t-rufang) [inline src/main/java/com/microsoft/azure/spark/tools/http/HttpObservable.java]: This is a pretty big change, lots of codes will be affected. Could you help to explain why we need this change?

**Comment by t-rufang** [comment]
> public class LivySparkBatchFactory implements SparkBatchJobFactory, Logger {

Seems like this `Logger` interface is not used?

---
Refers to: src/main/java/com/microsoft/azure/spark/tools/job/LivySparkBatchFactory.java:18 in 19ae439. [](commit_id = 19ae4390854724f52f331b3b2ccb4cc067908e5d, deletion_comment = False)

File: src/main/java/com/microsoft/azure/spark/tools/job/LivySparkBatchFactory.java [hunk 3]
```diff
@@ -62,6 +62,9 @@ public LivySparkBatchFactory controlSubject(final Observer<Pair<MessageInfoType,
         logAsCtrlSubject.subscribe(
                 typedMessage -> {
                     switch (typedMessage.getFirst()) {
+                        case Debug:
+                            log.debug(typedMessage.getRight());
```
Comment (t-rufang) [inline src/main/java/com/microsoft/azure/spark/tools/job/LivySparkBatchFactory.java]: Seems like the implemented interface `Logger` is not used?

Comment (wezhang) [reply src/main/java/com/microsoft/azure/spark/tools/job/LivySparkBatchFactory.java]: Fix it.

**Pull request merged**